### PR TITLE
Linear Interpolation

### PR DESCRIPTION
**Pull request by jduerholt**
This PR introduces GPs that are perfoming on linear interpolated two one-dimensional curves.

### CODE REVIEW

**Review by bertiqwerty — CHANGES_REQUESTED** [review]
Hi Johannes, Thanks. Can you please add some documentation? And an example would be great. Thanks.

**Review by bertiqwerty — APPROVED** [review]
After discussing with Johannes we decided to first merge this. Johannes will a more extensive documentation afterwards.

File: bofire/kernels/shape.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        x1: Tensor,
+        x2: Tensor,
+        diag: bool = False,
+        last_dim_is_batch: bool = False,
+    ) -> Tensor:
+        dists = self.calc_wasserstein_distances(x1, x2)
+        dists = dists / self.lengthscale
+        if self.squared:
+            return torch.exp(-(dists**2))
+        return torch.exp(-dists)
```
[the remark this answers is not in the document]
Comment (jduerholt) [reply bofire/kernels/shape.py:27]: We use it for monotonicall increasing 1 dimensional functions. On these functions, the wasserstein distance is equal to the integral over the absolute difference between the two functions.

[the remark this answers is not in the document]
Comment (CompRhys) [reply bofire/kernels/shape.py:27]: I would think that that should be explained in the docstring or rename to call it a laplacian_kernel. I was recently reading https://link.springer.com/article/10.1007/s10472-023-09883-w and so was drawn to this class but it's not as general as the name suggests.

[the remark this answers is not in the document]
Comment (jduerholt) [reply bofire/kernels/shape.py:27]: I will add it in the docstring. Why laplacian kernel?

[the remark this answers is not in the document]
Comment (CompRhys) [reply bofire/kernels/shape.py:27]: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.laplacian_kernel.html the cdist p=1 returns `|x-y||_1` which is the same distance as used in the laplacian_kernel although the prefactors and constants are different.

**Pull request merged**